client/src/components/StoreList/StoreList.tsx

```diff
@@ -20,13 +21,17 @@ function StoreList({ stores }: { stores: Store[] }) {
   return (
     <div className="StoreList">
       {stores &&
-        stores.map((store) => (
-          <StoreCard
-            key={store["store-id"]}
-            redirect={enterStore}
-            store={store}
-          />
-        ))}
+        stores.map((store: Store | null, i: number) =>
```

Comment:
same here, avoid null and keep StoreList rendering and GhostLoading animation separate on a top level component.

Reply:
added ghostloading components `PlaceholderStoreList` and `PlaceholderCart`